### ISSUE TITLE
Enable site logo alt text for accessibility

### ISSUE THREAD

**Issue by lwangdu** [WordPress/gutenberg#37104]
## What problem does this address?
<!--
When I add site logo, I can't add alt text.
![Screen Shot 2021-12-03 at 7 22 30 AM](https://user-images.githubusercontent.com/24829573/144628702-97467f50-ec09-425e-8fc6-791e1b47c199.png)

## What is your proposed solution?
It should be able to add alt text. This could be huge issue for SEO and accessibility.

**Comment by skorasaurus** [comment]
Great point; background - https://www.deque.com/blog/great-alt-text-introduction/

**Comment by Andrew-Starr** [comment]
The site logo already has alt text.

The default is the Site Title, unless overridden by entering a different alt text by editing the uploaded logo in the media library.

**Comment by lwangdu** [comment]
I understand that when I upload site logo in media library you can add the alt text.  After I added the site log on the header, there is no place that I can update or add alt text. If you could enable that, it would be great. 
Thank you.

**Comment by Andrew-Starr** [comment]
@lwangdu thanks for the additional information. The original comment was lacking detail so not too clear what the issue was.

**Comment by mikeybinns** [comment]
Just to add my point of view as a user of the block, it's not possible to set the alt text on use of the site logo block (like you can with an image block).

This means I cannot use the block in multiple places with different alt text, and also means that if I use the image in the media library in another location without setting the alt text, it will have the same alt text as the site-logo.

While this may not be an issue in most cases as it's always going to be the same image, one key case where it causes an issue is I may want to set different alt text if the logo is a link compared to when it isn't a link.

For example:
When it is linked to the homepage: `My company name. Link goes to the home page.`
When it is not linked: `My company name.`

**Comment by carolinan** [comment]
The only way I can think of, that uses the media library alt text by default, lets the user edit the alt text, and lets the user keep the alt text blank, is to use a toggle where the user chooses between the media library alt text and the customized alt text.
-And I think that is a bad user experience. Hopefully someone else can come up with something better.

**Comment by carolinan** [comment]
Related:
https://github.com/WordPress/gutenberg/issues/57893

**Comment by mikeybinns** [comment]
What if it pre-filled the alt text value when you insert the block into the page, but from there, it's its own field? You could optionally have a button to restore to default (the alt text of the media library file)

**Comment by joedolson** [comment]
This issue has gotten a lot of discussion recently in https://core.trac.wordpress.org/ticket/63838

**Comment by vishalkakadiya** [comment]
Another perspective on this feature is that the Site Logo block is typically not intended for repeated use across multiple pages. In scenarios where the logo needs to be reused with different alt text, the Image block can be used as an alternative, as it provides explicit support for customizable alt attributes.

@mikeybinns WDYT about my point?

**Comment by mikeybinns** [comment]
I disagree, I see the site logo block as a single image defined in site options, then used in multiple places. If it was separate and then I wanted to change the logo, I could no longer update the logo in a single place. The alt text change for me is less about describing the logo itself but providing context around the linking of the logo, which can be different per block instance.
